### DISCUSSION TITLE
Split properties for SARIF result

### DISCUSSION THREAD

**Discussion by nvuillam** [aquasecurity/trivy#4442] [Ideas]
Hi,

Thanks for your great tool ! 

As SARIF format supports `properties` with any key / value in it, would it be possible, to make a result more exploitable, to add the following values in those extra properties ?

https://github.com/aquasecurity/trivy/blob/11f4f811236ca05cfb30827e25f312eec45ca097/pkg/report/sarif.go#L143

Properties could be the ones not already present as standalone value on a result item:
- packageName
- installedVersion
- fixedVersion

```go
message: fmt.Sprintf("Package: %v\nInstalled Version: %v\nVulnerability %v\nSeverity: %v\nFixed Version: %v\nLink: [%v](%v)",
					vuln.PkgName, vuln.InstalledVersion, vuln.VulnerabilityID, vuln.Severity, vuln.FixedVersion, vuln.VulnerabilityID, vuln.PrimaryURL)
```

Many thanks & best regards :)

**Comment by afdesk** [comment]
@nvuillam thanks for your interest in `trivy`.
we'll definitely consider this suggestion.

**Comment by nvuillam** [comment]
Many thanks ! :)

**Comment by nvuillam** [comment]
it seems that SARIF format returns many less details than the JSON one :/

![image](https://user-images.githubusercontent.com/17500430/155387713-1f0d7030-5675-4d64-8c25-06bdc2b2fb45.png)

Would it be possible to also return in "properties" SARIF item everything that is returned by JSON result ? thanks :)

![image](https://user-images.githubusercontent.com/17500430/155387820-f3a5086d-3e83-4abb-9a97-de1eefb2cdf8.png)

Also, as we woud need that soon, would it be possible to have an estimated time for delivery of this feature ?
Or if you have not enough bandwith, I coudl eventually make my first PR is GO ?

**Comment by afdesk** [comment]
@nvuillam sorry, i missed your comment. 

> Or if you have not enough bandwith, I coudl eventually make my first PR is GO ?

it would be great! we welcome contributors!

**Comment by nvuillam** [comment]
Not stale :(

**Comment by nvuillam** [comment]
Still not stale :(

**Comment by nvuillam** [comment]
And again not stale 🤡

**Comment by nvuillam** [comment]
:(
